Domain/Cell/CellFactory.cs

```diff
@@ -0,0 +1,37 @@
+using DPAT_eindopdracht.Domain.Cell.State;
+
+namespace DPAT_eindopdracht.Domain.Cell;
+
+public class CellFactory
+{
+    //String defines the state of the cell
+    private Dictionary<string, Cell> cellTypes;
+
+    public CellFactory()
+    {
+        Cell emptyCell = new Cell(null);
+        emptyCell.SetState("empty");
+        Cell incorrectCell = new Cell(null);
+        incorrectCell.SetState("incorrect");
+        Cell correctCell = new Cell(null);
+        correctCell.SetState("correct");
+        
+        cellTypes = new Dictionary<string, Cell>()
+        {
+            {"empty", emptyCell},
+            {"incorrect", incorrectCell},
+            {"correct", correctCell}
+        };
```

Comment:
Misschien in een losse functie Create Prototypes() Zodat het duidelijk is dat dit gebruikt wordt voor prototype pattern.